tests/core/test_device_collector.py

```diff
@@ -56,19 +61,69 @@ def test_device_connector_sync_run_engine_created_connects(RE):
     assert working_device.connected
 
 
+def test_device_connector_async_run_engine_same_event_loop():
+    async def set_up_device():
+        async with DeviceCollector(sim=True):
+            sim_motor = motor.Motor("BLxxI-MO-TABLE-01:X")
+        return sim_motor
+
+    loop = asyncio.new_event_loop()
+    checking_loop = asyncio.new_event_loop()
+
+    try:
+        sim_motor = loop.run_until_complete(set_up_device())
+        RE = RunEngine(call_returns_result=True, loop=loop)
+
+        def my_plan():
+            yield from bps.mov(sim_motor, 3.14)
+
+        RE(my_plan())
+
+        assert (
+            checking_loop.run_until_complete(sim_motor.setpoint.read())[
+                "sim_motor-setpoint"
+            ]["value"]
+            == 3.14
+        )
+
+    finally:
+        if RE.state not in ("idle", "panicked"):
+            try:
+                RE.halt()
+            except TransitionError:
+                pass
+        loop.call_soon_threadsafe(loop.stop)
+        checking_loop.call_soon_threadsafe(checking_loop.stop)
+        RE._th.join()
+        loop.close()
+        checking_loop.close()
+
+
 """
-# TODO: Once passing a loop into the run-engine selector works, this should pass
-async def test_device_connector_async_run_engine_same_event_loop():
-    async with DeviceCollector(sim=True):
-        sim_motor = motor.Motor("BLxxI-MO-TABLE-01:X")
+# SimSignalBackend currently allows a different event loop to set the value
+# so this test is not working
+def test_device_connector_async_run_engine_different_event_loop():
```

Comment:
Would prefer `@pytest.skip(reason="<link to github issue>")` rather than commenting the whole test out.